src/components/Cluster/ClusterDetail/AppDetailsModal/AppDetailsModal.js

```diff
@@ -0,0 +1,254 @@
+import { deleteApp, loadApps } from 'actions/appActions';
+import {
+  createAppConfig,
+  deleteAppConfig,
+  updateAppConfig,
+} from 'actions/appConfigActions';
+import {
+  createAppSecret,
+  deleteAppSecret,
+  updateAppSecret,
+} from 'actions/appSecretActions';
+import GenericModal from 'components/Modals/GenericModal';
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import Button from 'UI/Button';
+import ClusterIDLabel from 'UI/ClusterIDLabel';
+
+import EditChartVersionPane from './EditChartVersionPane';
+import InitialPane from './InitialPane';
+
+const modalPanes = {
+  deleteAppConfig: 'deleteAppConfig',
+  deleteAppSecret: 'deleteAppSecret',
+  deleteApp: 'deleteApp',
+  editChartVersion: 'editChartVersion',
+  initial: 'initial',
+};
+
+const AppDetailsModal = props => {
+  const [pane, setPane] = useState(modalPanes.initial);
+
+  if (props.app === null || typeof props.app === 'undefined') {
+    return <span />;
+  }
+
+  const appName = props.app.metadata.name;
+  const clusterId = props.clusterId;
+
+  function showPane(paneToShow) {
+    return function() {
+      setPane(paneToShow);
+    };
+  }
+
+  function onClose() {
+    showPane(modalPanes.initial)();
+    props.onClose();
+  }
+
+  async function loadAppsAndClose() {
+    await props.dispatch(loadApps(clusterId));
+    onClose();
+  }
+
+  async function _deleteAppConfig() {
+    await props.dispatch(deleteAppConfig(appName, clusterId));
+    await loadAppsAndClose();
+  }
+
+  async function _deleteAppSecret() {
+    await props.dispatch(deleteAppSecret(appName, clusterId));
+    await loadAppsAndClose();
+  }
+
+  async function _deleteApp() {
+    await props.dispatch(deleteApp(appName, clusterId));
+    await loadAppsAndClose();
+  }
+
+  async function _createAppConfig(values, done) {
+    await props.dispatch(createAppConfig(appName, clusterId, values));
+    await loadAppsAndClose();
+    done();
+  }
+
+  async function _updateAppConfig(values, done) {
+    await props.dispatch(updateAppConfig(appName, clusterId, values));
+    await loadAppsAndClose();
+    done();
+  }
+
+  async function _createAppSecret(values, done) {
+    await props.dispatch(createAppSecret(appName, clusterId, values));
+    await loadAppsAndClose();
+    done();
+  }
+
+  async function _updateAppSecret(values, done) {
+    await props.dispatch(updateAppSecret(appName, clusterId, values));
+    await loadAppsAndClose();
+    done();
+  }
+
+  function deleteConfirmFooter(cta, onConfirm) {
```

Comment:
Can you please extract this to a different file?

Reply:
Right now it's a little jsx fragment that is repeated otherwise in a few places, I don't feel like it warrants being in it's own file / component. Do you feel strongly about it?

Reply:
Also resisting because then I end up with more lines due to how prettier does the new lines.

So it goes from this one liner:

```
modalFooter = deleteConfirmFooter('Delete App', _deleteApp);
```

to

```
modalFooter =  <DeleteConfirmFooter
        cta={Delete App}
        onConfirm={_deleteApp}
        onCancel={showPane(modalPanes.initial)}
/>
```

Reply:
I could keep the function and extract the jsx fragment if you think it makes things better.

```
  function deleteConfirmFooter(cta, onConfirm) {
    return (
      <DeleteConfirmFooter
        cta={cta}
        onConfirm={onConfirm}
        onCancel={showPane(modalPanes.initial)}
      />
    );
  }
```

Reply:
I like to have components in separate files, but not forcing it if you want to do otherwise. 
But I believe we should at least extract `deleteConfirmFooter()` from the main function body, because we're basically creating a new component with every update. If it was separate, react will be able to reuse it

Reply:
I saw now that you extracted it. The comment below is no longer applicable 👍